File: security-app/security-app/test-model.js
const { GoogleGenerativeAI } = require("@google/generative-ai");

const apiKey = "AIzaSyAtCtSmnxMGKS4UmWWOz40aVfRa2L_cd2w";
const genAI = new GoogleGenerativeAI(apiKey);

async function testModel(modelName) {
    console.log(`Testing model: ${modelName}`);
    try {
        const model = genAI.getGenerativeModel({ model: modelName });
        const result = await model.generateContent("Hello!");
        const response = await result.response;
        console.log(`Success! Response: ${response.text()}`);
        return true;
    } catch (error) {
        console.error(`Failed with ${modelName}:`, error.message);
        return false;
    }
}

async function main() {
    // Try a few likely candidates in order or preference
    const candidates = [
        "gemini-1.5-flash",
        "gemini-1.5-flash-001",
        "gemini-1.5-flash-002",
        "gemini-1.5-pro",
        "gemini-pro"
    ];

    for (const m of candidates) {
        if (await testModel(m)) {
            console.log(`\n>>> FOUND WORKING MODEL: ${m} <<<`);
            break;
        }
    }
}

main();

File: QAutoX AI/tests/test_model.py
"""
Test gemini-2.5-flash model
"""
import google.generativeai as genai
import os
from dotenv import load_dotenv

load_dotenv()
GEMINI_API_KEY = os.getenv('GEMINI_API_KEY')

print("Testing gemini-2.5-flash model...")
print("=" * 60)

try:
    genai.configure(api_key=GEMINI_API_KEY)
    model = genai.GenerativeModel('gemini-2.5-flash')
    
    response = model.generate_content(
        "Generate a simple JSON array with 2 test cases for a login feature. Return ONLY valid JSON.",
        generation_config={"response_mime_type": "application/json"}
    )
    
    print("✓ SUCCESS with gemini-2.5-flash!")
    print(f"Response length: {len(response.text)} characters")
    print(f"Response preview:\n{response.text[:400]}...")
    
except Exception as e:
    print(f"✗ ERROR: {type(e).__name__}")
    print(f"Error: {str(e)[:300]}")
